internal/extension/reconciler_test.go

```diff
@@ -4,8 +4,120 @@ import (
 	"sync"
 	"testing"
 	"time"
+
+	"github.com/kuadrant/policy-machinery/machinery"
+	"github.com/samber/lo"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gwapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	v0 "github.com/kuadrant/kuadrant-operator/pkg/extension/grpc/v0"
 )
 
+func TestStateAwareDAG(t *testing.T) {
```

Comment:
This is from `policy-machinery`... 😢 
It just is really tedious to get a dag and I didn't know how to reuse that bit... so `Ctrl+C` & `Ctrl+V` myself outta that